File: leaoai_python_refatorado_descompactado/app/schemas/Factory/projecttemplate.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ProjectTemplateBase(BaseModel):
    Id: int
    Name: str
    TemplateJson: str
    Ativo: bool
    Excluido: bool
    CreatedAt: datetime

class ProjectTemplateCreate(ProjectTemplateBase):
    pass # Add specific fields for creation if needed

class ProjectTemplate(ProjectTemplateBase):
    Id: int

    class Config:
        from_attributes = True

File: Framework.Data/Models/Factory/ProjectTemplate.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Data.Models
{
    public class ProjectTemplate
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string TemplateJson { get; set; }

        public bool Ativo { get; set; } = true;
        public bool Excluido { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}
